### PR TITLE
finer level only granular diag dumps

### PR DESCRIPTION
**Pull request by Dekken**
https://github.com/PHAREHUB/PHARE/issues/359

```
h5ls -r tests/simulator/phare_outputs/diagnostic_test/1_2_mpi_n_1/EM_B.h5 
/                        Group
/t0.000000               Group
/t0.000000/pl0           Group
/t0.000000/pl0/p0#0      Group
/t0.000000/pl0/p0#0/EM_B_x Dataset {51}
/t0.000000/pl0/p0#0/EM_B_y Dataset {50}
/t0.000000/pl0/p0#0/EM_B_z Dataset {50}
/t0.000000/pl1           Group
/t0.000000/pl1/p0#0      Group
/t0.000000/pl1/p0#0/EM_B_x Dataset {33}
/t0.000000/pl1/p0#0/EM_B_y Dataset {32}
/t0.000000/pl1/p0#0/EM_B_z Dataset {32}
/t0.000100               Group
/t0.000100/pl1           Group
/t0.000100/pl1/p0#0      Group
/t0.000100/pl1/p0#0/EM_B_x Dataset {33}
/t0.000100/pl1/p0#0/EM_B_y Dataset {32}
/t0.000100/pl1/p0#0/EM_B_z Dataset {32}
/t0.000200               Group
/t0.000200/pl1           Group
/t0.000200/pl1/p0#0      Group
/t0.000200/pl1/p0#0/EM_B_x Dataset {33}
/t0.000200/pl1/p0#0/EM_B_y Dataset {32}
/t0.000200/pl1/p0#0/EM_B_z Dataset {32}
/t0.000300               Group
/t0.000300/pl1           Group
/t0.000300/pl1/p0#0      Group
/t0.000300/pl1/p0#0/EM_B_x Dataset {33}
/t0.000300/pl1/p0#0/EM_B_y Dataset {32}
/t0.000300/pl1/p0#0/EM_B_z Dataset {32}
/t0.000400               Group
/t0.000400/pl1           Group
/t0.000400/pl1/p0#0      Group
/t0.000400/pl1/p0#0/EM_B_x Dataset {33}
/t0.000400/pl1/p0#0/EM_B_y Dataset {32}
/t0.000400/pl1/p0#0/EM_B_z Dataset {32}
/t0.000500               Group
/t0.000500/pl1           Group
/t0.000500/pl1/p0#0      Group
/t0.000500/pl1/p0#0/EM_B_x Dataset {33}
/t0.000500/pl1/p0#0/EM_B_y Dataset {32}
/t0.000500/pl1/p0#0/EM_B_z Dataset {32}
/t0.000600               Group
/t0.000600/pl1           Group
/t0.000600/pl1/p0#0      Group
/t0.000600/pl1/p0#0/EM_B_x Dataset {33}
/t0.000600/pl1/p0#0/EM_B_y Dataset {32}
/t0.000600/pl1/p0#0/EM_B_z Dataset {32}
/t0.000700               Group
/t0.000700/pl1           Group
/t0.000700/pl1/p0#0      Group
/t0.000700/pl1/p0#0/EM_B_x Dataset {33}
/t0.000700/pl1/p0#0/EM_B_y Dataset {32}
/t0.000700/pl1/p0#0/EM_B_z Dataset {32}
/t0.000800               Group
/t0.000800/pl1           Group
/t0.000800/pl1/p0#0      Group
/t0.000800/pl1/p0#0/EM_B_x Dataset {33}
/t0.000800/pl1/p0#0/EM_B_y Dataset {32}
/t0.000800/pl1/p0#0/EM_B_z Dataset {32}
/t0.000900               Group
/t0.000900/pl1           Group
/t0.000900/pl1/p0#0      Group
/t0.000900/pl1/p0#0/EM_B_x Dataset {33}
/t0.000900/pl1/p0#0/EM_B_y Dataset {32}
/t0.000900/pl1/p0#0/EM_B_z Dataset {32}
/t0.001000               Group
/t0.001000/pl0           Group
/t0.001000/pl0/p0#0      Group
/t0.001000/pl0/p0#0/EM_B_x Dataset {51}
/t0.001000/pl0/p0#0/EM_B_y Dataset {50}
/t0.001000/pl0/p0#0/EM_B_z Dataset {50}
/t0.001000/pl1           Group
/t0.001000/pl1/p0#0      Group
/t0.001000/pl1/p0#0/EM_B_x Dataset {33}
/t0.001000/pl1/p0#0/EM_B_y Dataset {32}
/t0.001000/pl1/p0#0/EM_B_z Dataset {32}
```

### CODE REVIEW

**Comment by Dekken** [comment]
A cppdict may be sufficient as the "generic parameter" now that I think about it

File: src/solver/multiphysics_integrator.h [hunk 1]
```diff
@@ -394,6 +400,21 @@ namespace solver
             fromCoarser.prepareStep(model, *level);
 
 
+            // we skip first/last as that's done via regular diag dump mechanism
+            bool fineDumpsActive = simFuncs_.at("pre_advance").count("fine_dump") > 0;
+            bool notCoarestTime  = currentTime != firstNewLevelTimes_[0];
```
Comment (nicolasaunai) [inline src/solver/multiphysics_integrator.h:405]: notCoarsest

File: tests/simulator/diagnostics.py [hunk 2]
```diff
@@ -156,18 +172,68 @@ def _test_dump_diags(self, dim, **simInput):
...
+
+
+
+    _has_patch_ghost_on_refined_level_case = (
+      dup({
+        "smallest_patch_size": 5,
+        "largest_patch_size": 5}),
+    )
+    @data(*_has_patch_ghost_on_refined_level_case)
+    def test_has_patch_ghost_on_refined_level_case(self, simInput):
```
Comment (nicolasaunai) [inline tests/simulator/diagnostics.py:234]: aren't these init tests ?

Comment (Dekken) [reply tests/simulator/diagnostics.py:234]: we're doing direct h5py access, so maybe not, or maybe there's another way via the hierarchy

File: tests/simulator/test_initialization.py [hunk 3]
```diff
@@ -656,6 +666,60 @@ def test_levelghostparticles_have_correct_split_from_coarser_particle(self, refi
...
+        import pyphare.pharein as ph
+        from pybindlibs import cpp
+        from pyphare.simulator.simulator import startMPI
+
+        startMPI()
+
+        out = "phare_outputs"
+
+        test_id = self.ddt_test_id()
+        if cpp.mpi_size() > 1: # SKIP
```
Comment (nicolasaunai) [inline tests/simulator/test_initialization.py:682]: could you comment on why this test is skipped for parallel jobs?

Comment (Dekken) [reply tests/simulator/test_initialization.py:682]: yes, haven't spent much time on figuring it out tbh

File: tests/simulator/diagnostics.py (hunk 2 above)
Comment (Dekken) [reply tests/simulator/diagnostics.py]: moved to test_initlization.py

**Review by nicolasaunai — APPROVED** [review]

**Pull request merged**